src/Momento.Sdk/Config/Transport/StaticTransportStrategy.cs

```diff
@@ -27,13 +34,27 @@ public class StaticGrpcConfiguration : IGrpcConfiguration
     /// <param name="grpcChannelOptions">Customizations to low-level gRPC channel configuration</param>
     /// <param name="minNumGrpcChannels">minimum number of gRPC channels to open</param>
     /// <param name="socketsHttpHandlerOptions">Customizations to the SocketsHttpHandler</param>
-    public StaticGrpcConfiguration(TimeSpan deadline, GrpcChannelOptions? grpcChannelOptions = null, int minNumGrpcChannels = 1, SocketsHttpHandlerOptions? socketsHttpHandlerOptions = null)
+    /// <param name="keepAlivePingTimeout">Override the time to wait for a response from a keepalive ping</param>
+    /// <param name="keepAlivePingDelay">Override how often the client sends keepalive pings the server</param>
+    /// <param name="keepAlivePermitWithoutCalls">Sets option to send keepalive pings from the client without any outstanding RPCs</param>
+    public StaticGrpcConfiguration(
+        TimeSpan deadline, 
+        GrpcChannelOptions? grpcChannelOptions = null, 
+        int minNumGrpcChannels = 1, 
+        SocketsHttpHandlerOptions? socketsHttpHandlerOptions = null,
+        TimeSpan? keepAlivePingTimeout = null, 
+        TimeSpan? keepAlivePingDelay = null, 
+        bool? keepAlivePermitWithoutCalls = null
+    )
     {
         Utils.ArgumentStrictlyPositive(deadline, nameof(deadline));
         this.Deadline = deadline;
         this.MinNumGrpcChannels = minNumGrpcChannels;
-        this.GrpcChannelOptions = grpcChannelOptions ?? new GrpcChannelOptions();
+        this.GrpcChannelOptions = grpcChannelOptions;
         this.SocketsHttpHandlerOptions = socketsHttpHandlerOptions ?? new SocketsHttpHandlerOptions();
+        this.KeepAlivePingTimeout = keepAlivePingTimeout ?? System.Threading.Timeout.InfiniteTimeSpan;
+        this.KeepAlivePingDelay = keepAlivePingDelay ?? System.Threading.Timeout.InfiniteTimeSpan;
+        this.KeepAlivePermitWithoutCalls = keepAlivePermitWithoutCalls ?? false;
```

Comment:
couldn't find a way to optionally set or omit each keepalive setting when creating the `SocketsHttpHandler` in the `Utils.GrpcChannelOptionsFromGrpcConfig` function, so this part will set each one to the config-provided value or to a default value ([delay](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.keepalivepingdelay?view=net-6.0#system-net-http-socketshttphandler-keepalivepingdelay) and [timeout](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.keepalivepingtimeout?view=net-6.0#system-net-http-socketshttphandler-keepalivepingtimeout) are disabled when set to infinite time span, policy defaults to [WithActiveRequests](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpkeepalivepingpolicy?view=net-6.0))

Reply:
i think this will shake out more cleanly when you get rid of `FromGrpcConfig`and put these new config options into the SocketHandlerOptions section as you suggested above?